app/forms/punches_create_form.rb

```diff
@@ -0,0 +1,46 @@
+class PunchesCreateForm
+  include ActiveModel::Model
+
+  attr_accessor :punches
+
+  validates :punches, presence: true
+  validate :punches_validations, :punch_on_same_day_validation
+
+  def initialize(params)
+    params ||= {}
+    params[:punches] ||= []
+    params[:user] ||= User.new
+
+    @punches = params[:punches].map do |punch_params|
+      Punch.new(**punch_params, user: params[:user], company: params[:user].company)
+    end
+  end
+
+  def save
+    return false unless valid?
+
+    @punches = Punch.transaction do
+      @punches.each(&:save!)
+    end
+
+    true
+  end
+
+  private
+
+  def punches_validations
+    @punches.each do |punch|
+      errors.merge!(punch.errors) if punch.invalid?
+    end
+  end
+
+  def punch_on_same_day_validation
+    @punches.each do |punch|
+      errors.add(:from, 'There is already a punch on the same day') if there_punch_on_the_same_day?(punch)
+    end
+  end
+
+  def there_punch_on_the_same_day?(punch)
```

Comment:
```suggestion
  def multiple_punches_on_the_same_day?(punch)
```
I think this method could have a different name for better readability.

Reply:
I don't thing this is the idea, the function wanna know if there are at least 1 other punch already on database, not if have multiples

Reply:
In that case you can call it `has_punches_on_the_same_day?`

Also, the function itself doesn't check for a single punch, but for multiple punches (which is the correct validation), so I think using `punches` instead of `punch` in the method name makes more sense.